### PR TITLE
Veselov Ilya. Lab 4. Option 4.

### PR DESCRIPTION
**Pull request by spells39**

### CODE REVIEW

**Comment by aobolensk** [comment]
It is lab 3, isn't it?

**Comment by spells39** [comment]
> It is lab 3, isn't it?

No, this is the fourth laboratory work. It's just that for some reason the third one got into this PR.

**Comment by aobolensk** [comment]
> > It is lab 3, isn't it?
> 
> 
> 
> No, this is the fourth laboratory work. It's just that for some reason the third one got into this PR.

Please, remove it from this PR

File: mlir/lib/Transforms/lab4/veselov_ilya/VeselovIlyaCounterFuncCalls.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  StringRef getDescription() const final {
+    return "Counts amount of times it was called by other functions in the "
+           "module";
+  }
+
+  void runOnOperation() override {
+    ModuleOp mod = getOperation();
+    std::map<StringRef, int> calls;
+    mod.walk([&](Operation *oper) {
+      if (auto callOper = dyn_cast<LLVM::CallOp>(oper)) {
```
Comment (aobolensk) [inline mlir/lib/Transforms/lab4/veselov_ilya/VeselovIlyaCounterFuncCalls.cpp]: No need to use walker that walks through all operations and then filter. Walker can filter by itself

File: mlir/test/Transforms/lab4/veselov_ilya/test.mlir [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// RUN: mlir-opt -load-pass-plugin=%mlir_lib_dir/VeselovIlyaCounterFuncCalls%shlibext --pass-pipeline="builtin.module(VeselovIlyaCounterFuncCalls)" %s | FileCheck %s
+
+module attributes {dlti.dl_spec = #dlti.dl_spec<#dlti.dl_entry<f16, dense<16> : vector<2xi32>>, #dlti.dl_entry<f64, dense<64> : vector<2xi32>>, #dlti.dl_entry<i16, dense<16> : vector<2xi32>>, #dlti.dl_entry<i32, dense<32> : vector<2xi32>>, #dlti.dl_entry<i8, dense<8> : vector<2xi32>>, #dlti.dl_entry<!llvm.ptr, dense<64> : vector<4xi32>>, #dlti.dl_entry<i1, dense<8> : vector<2xi32>>, #dlti.dl_entry<f80, dense<128> : vector<2xi32>>, #dlti.dl_entry<i64, dense<64> : vector<2xi32>>, #dlti.dl_entry<!llvm.ptr<270>, dense<32> : vector<4xi32>>, #dlti.dl_entry<f128, dense<128> : vector<2xi32>>, #dlti.dl_entry<!llvm.ptr<272>, dense<64> : vector<4xi32>>, #dlti.dl_entry<!llvm.ptr<271>, dense<32> : vector<4xi32>>, #dlti.dl_entry<"dlti.endianness", "little">, #dlti.dl_entry<"dlti.stack_alignment", 128 : i32>>} {
```
Comment (aobolensk) [inline mlir/test/Transforms/lab4/veselov_ilya/test.mlir]:
```suggestion
module {
```

File: mlir/test/Transforms/lab4/veselov_ilya/test.mlir [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  llvm.func @_Z5emptyv() attributes {passthrough = ["mustprogress", "noinline", "nounwind", "optnone", ["uwtable", "2"], ["frame-pointer", "all"], ["min-legal-vector-width", "0"], ["no-trapping-math", "true"], ["stack-protector-buffer-size", "8"], ["target-cpu", "x86-64"], ["target-features", "+cx8,+fxsr,+mmx,+sse,+sse2,+x87"], ["tune-cpu", "generic"]]} {
+    // CHECK: countOfCalls = 2
```
Comment (aobolensk) [inline mlir/test/Transforms/lab4/veselov_ilya/test.mlir]: Please match function name as a part of this check as well

**Review by aobolensk — CHANGES_REQUESTED** [review]

**Review by m-ly4 — APPROVED** [review]

**Pull request merged**